packages/rocketchat-slackbridge/slackbridge.js

```diff
@@ -0,0 +1,712 @@
+			this.userTags[userId] = { slack: `<@${userId}>`, rocket: `@${user.username}` };
+		}
+		return user;
+	}
+
+	addUser(userId) {
+		let data = HTTP.get('https://slack.com/api/users.info', { params: { token: this.apiToken, user: userId } });
+		if (data && data.data && data.data.ok === true && data.data.user && data.data.user.profile && data.data.user.profile.email) {
+			let userData = data.data.user;
+			let existingUser = RocketChat.models.Users.findOneByEmailAddress(userData.profile.email) || RocketChat.models.Users.findOneByUsername(userData.name);
+			if (existingUser) {
+				userData.rocketId = existingUser._id;
+				userData.name = existingUser.username;
+			} else {
+				userData.rocketId = Accounts.createUser({ email: userData.profile.email, password: Date.now() + userData.name + userData.profile.email.toUpperCase() });
+				Meteor.runAsUser(userData.rocketId, () => {
+					Meteor.call('setUsername', userData.name);
+					Meteor.call('joinDefaultChannels', true);
+					let url = null;
+					if (userData.profile.image_original) {
+						url = userData.profile.image_original;
+					} else if (userData.profile.image_512) {
+						url = userData.profile.image_512;
+					}
+					Meteor.call('setAvatarFromService', url, null, 'url');
+					// Slack's is -18000 which translates to Rocket.Chat's after dividing by 3600
+					if (userData.tz_offset) {
+						Meteor.call('updateUserUtcOffset', userData.tz_offset / 3600);
+					}
+					if (userData.profile.real_name) {
+						RocketChat.models.Users.setName(userData.rocketId, userData.profile.real_name);
+						// Deleted users are 'inactive' users in Rocket.Chat
+						if (userData.deleted) {
+							Meteor.call('setUserActiveStatus', userData.rocketId, false);
+						}
+					}
+				});
+			}
+			RocketChat.models.Users.update({ _id: userData.rocketId }, { $addToSet: { importIds: userData.id } });
+			if (!this.userTags[userId]) {
+				this.userTags[userId] = { slack: `<@${userId}>`, rocket: `@${userData.name}` };
+			}
+			return RocketChat.models.Users.findOneById(userData.rocketId);
+		}
+
+		return;
+	}
+
+	saveMessage(room, user, message, msgDataDefaults) {
+		if (message.type === 'message') {
+			let msgObj = {};
+			if (!_.isEmpty(message.subtype)) {
+				msgObj = this.processSubtypedMessage(room, user, message, msgDataDefaults);
+				if (!msgObj) {
+					return;
+				}
+			} else {
+				msgObj = {
+					msg: this.convertSlackMessageToRocketChat(message.text),
+					rid: room._id,
+					u: {
+						_id: user._id,
+						username: user.username
+					}
+				};
+			}
+			_.extend(msgObj, msgDataDefaults);
+			if (message.edited) {
+				msgObj.ets = new Date(parseInt(message.edited.ts.split('.')[0]) * 1000);
+			}
+			if (message.subtype === 'bot_message') {
+				user = RocketChat.models.Users.findOneById('rocket.cat', { fields: { username: 1 } });
+			}
+			RocketChat.sendMessage(user, msgObj, room, { upsert: true });
+		}
+	}
+
+	processSubtypedMessage(room, user, message) {
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [This function's cyclomatic complexity is too high. (30)](https://www.codacy.com/app/RocketChat/Rocket-Chat/file/2677595092/issues/source?bid=3199634&fileBranchId=3334129#l232)